### ISSUE TITLE
Missing X server or $DISPLAY

### ISSUE THREAD

**Issue by marmalodak** [Heroic-Games-Launcher/HeroicGamesLauncher#2317]
### Describe the bug

Instead of launching, heroic reports "core dumped".

### Add logs

```text
❯ heroic
[41583:0105/214726.096362:ERROR:ozone_platform_x11.cc(240)] Missing X server or $DISPLAY
[41583:0105/214726.096394:ERROR:env.cc(255)] The platform failed to initialize.  Exiting.
The futex facility returned an unexpected error code.
[1]    41583 IOT instruction (core dumped)  heroic


           PID: 38447 (heroic)
           UID: 1000 (john)
           GID: 1000 (john)
        Signal: 6 (ABRT)
     Timestamp: Thu 2023-01-05 21:28:11 PST (11min ago)
  Command Line: $'/usr/lib64/heroic-games-launcher-bin/heroic --help --enable-crashpad'
    Executable: /usr/lib64/heroic-games-launcher-bin/heroic
 Control Group: /user.slice/user-1000.slice/session-2.scope
          Unit: session-2.scope
         Slice: user-1000.slice
       Session: 2
     Owner UID: 1000 (john)
       Boot ID: 2dfad6ec15bb401a97384d6aa2641436
    Machine ID: a533d46ed2fe4e02bf799c6ddae034a8
      Hostname: mediapc
       Storage: /var/lib/systemd/coredump/core.heroic.1000.2dfad6ec15bb401a97384d6aa2641436.38447.1672982891000000.zst (present)
     Disk Size: 13.2M
       Message: Process 38447 (heroic) of user 1000 dumped core.
                
                Module linux-vdso.so.1 with build-id 1e1aa6d2fac3d10224e5832a9bc8a2c1ba3b085b
                Module libbrotlicommon.so.1 with build-id e2790c03a5c688b7e75e89676cdd2b5fcf247a6f
                Metadata for module libbrotlicommon.so.1 owned by FDO found: {
                	"type" : "rpm",
                	"name" : "brotli",
                	"version" : "1.0.9-7.fc36",
                	"architecture" : "x86_64",
                	"osCpe" : "cpe:/o:fedoraproject:fedora:36"
                }
                
                Module libstdc++.so.6 with build-id 6ce2a41dee893f9ede3c63f9aadad3826220476e
                Module libicudata.so.69 with build-id 4d19cd548e77f2f9778d81cf9c27831200804bc4
                Stack trace of thread 38447:
                #0  0x00007f008ae8ec0c __pthread_kill_implementation (libc.so.6 + 0x8ec0c)
                #1  0x00007f008ae3e986 raise (libc.so.6 + 0x3e986)
                #2  0x00007f008ae287f4 abort (libc.so.6 + 0x287f4)
                #3  0x00007f008ae82d5e __libc_message (libc.so.6 + 0x82d5e)
                #4  0x00007f008ae82dda __libc_fatal (libc.so.6 + 0x82dda)
                #5  0x00007f008ae944cd sem_post@GLIBC_2.2.5 (libc.so.6 + 0x944cd)
                #6  0x000055ce22768027 uv_sem_post (heroic + 0x2213027)
                #7  0x000055ce228a9e47 n/a (heroic + 0x2354e47)
                #8  0x000055ce228da55e n/a (heroic + 0x238555e)
                #9  0x000055ce2282c560 n/a (heroic + 0x22d7560)
                #10 0x000055ce2282c63e n/a (heroic + 0x22d763e)
                #11 0x000055ce24937cfb n/a (heroic + 0x43e2cfb)
                #12 0x000055ce24937e1e n/a (heroic + 0x43e2e1e)
                #13 0x000055ce2493c088 n/a (heroic + 0x43e7088)
                #14 0x000055ce2493bb58 n/a (heroic + 0x43e6b58)
                #15 0x000055ce2493796f n/a (heroic + 0x43e296f)
                #16 0x000055ce229f7ab1 n/a (heroic + 0x24a2ab1)
                #17 0x000055ce229f8d9c n/a (heroic + 0x24a3d9c)
                #18 0x000055ce229f8b91 n/a (heroic + 0x24a3b91)
                #19 0x000055ce229f650f n/a (heroic + 0x24a150f)
                #20 0x000055ce229f6624 n/a (heroic + 0x24a1624)
                #21 0x000055ce2276d7cb n/a (heroic + 0x22187cb)
                #22 0x00007f008ae29510 __libc_start_call_main (libc.so.6 + 0x29510)
                #23 0x00007f008ae295c9 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x295c9)
                #24 0x000055ce223d802a _start (heroic + 0x1e8302a)
                
                Stack trace of thread 38461:
                #0  0x00007f008af1194e epoll_wait (libc.so.6 + 0x11194e)
                #1  0x000055ce2276adf4 n/a (heroic + 0x2215df4)
                #2  0x000055ce2275c782 uv_run (heroic + 0x2207782)
                #3  0x000055ce293cf7b1 n/a (heroic + 0x8e7a7b1)
                #4  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #5  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38455:
                #0  0x00007f008af1194e epoll_wait (libc.so.6 + 0x11194e)
                #1  0x000055ce25945136 n/a (heroic + 0x53f0136)
                #2  0x000055ce25942e45 n/a (heroic + 0x53ede45)
                #3  0x000055ce2581906c n/a (heroic + 0x52c406c)
                #4  0x000055ce257cad7c n/a (heroic + 0x5275d7c)
                #5  0x000055ce2578fb72 n/a (heroic + 0x523ab72)
                #6  0x000055ce257e82e8 n/a (heroic + 0x52932e8)
                #7  0x000055ce257d3e1d n/a (heroic + 0x527ee1d)
                #8  0x000055ce257e8479 n/a (heroic + 0x5293479)
                #9  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #10 0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #11 0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38460:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c1d0 pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x8c1d0)
                #2  0x000055ce25807381 n/a (heroic + 0x52b2381)
                #3  0x000055ce25807b31 n/a (heroic + 0x52b2b31)
                #4  0x000055ce258078ce n/a (heroic + 0x52b28ce)
                #5  0x000055ce2576f85c n/a (heroic + 0x521a85c)
                #6  0x000055ce257cad7c n/a (heroic + 0x5275d7c)
                #7  0x000055ce2578fb72 n/a (heroic + 0x523ab72)
                #8  0x000055ce257e82e8 n/a (heroic + 0x52932e8)
                #9  0x000055ce257e8479 n/a (heroic + 0x5293479)
                #10 0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #11 0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #12 0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38465:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae94c50 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0x94c50)
                #2  0x000055ce22768090 uv_sem_wait (heroic + 0x2213090)
                #3  0x000055ce2944086d n/a (heroic + 0x8eeb86d)
                #4  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #5  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38464:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c1d0 pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x8c1d0)
                #2  0x000055ce22768249 uv_cond_wait (heroic + 0x2213249)
                #3  0x000055ce293cf95a n/a (heroic + 0x8e7a95a)
                #4  0x000055ce293cd5a2 n/a (heroic + 0x8e785a2)
                #5  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #6  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38456:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c4c4 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x8c4c4)
                #2  0x000055ce258074cb n/a (heroic + 0x52b24cb)
                #3  0x000055ce25807b02 n/a (heroic + 0x52b2b02)
                #4  0x000055ce257e13ec n/a (heroic + 0x528c3ec)
                #5  0x000055ce257e1f1d n/a (heroic + 0x528cf1d)
                #6  0x000055ce257e1b6d n/a (heroic + 0x528cb6d)
                #7  0x000055ce257e1a81 n/a (heroic + 0x528ca81)
                #8  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #9  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #10 0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38457:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c4c4 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x8c4c4)
                #2  0x000055ce258074cb n/a (heroic + 0x52b24cb)
                #3  0x000055ce25807b02 n/a (heroic + 0x52b2b02)
                #4  0x000055ce257e13ec n/a (heroic + 0x528c3ec)
                #5  0x000055ce257e1f1d n/a (heroic + 0x528cf1d)
                #6  0x000055ce257e1b6d n/a (heroic + 0x528cb6d)
                #7  0x000055ce257e1a81 n/a (heroic + 0x528ca81)
                #8  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #9  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #10 0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38459:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c4c4 pthread_cond_timedwait@@GLIBC_2.3.2 (libc.so.6 + 0x8c4c4)
                #2  0x000055ce258074cb n/a (heroic + 0x52b24cb)
                #3  0x000055ce25807b02 n/a (heroic + 0x52b2b02)
                #4  0x000055ce257e13ec n/a (heroic + 0x528c3ec)
                #5  0x000055ce257e1cf6 n/a (heroic + 0x528ccf6)
                #6  0x000055ce257e1b6d n/a (heroic + 0x528cb6d)
                #7  0x000055ce257e1a81 n/a (heroic + 0x528ca81)
                #8  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #9  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #10 0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38462:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c1d0 pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x8c1d0)
                #2  0x000055ce22768249 uv_cond_wait (heroic + 0x2213249)
                #3  0x000055ce293cf95a n/a (heroic + 0x8e7a95a)
                #4  0x000055ce293cd5a2 n/a (heroic + 0x8e785a2)
                #5  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #6  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38463:
                #0  0x00007f008ae899d9 __futex_abstimed_wait_common (libc.so.6 + 0x899d9)
                #1  0x00007f008ae8c1d0 pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x8c1d0)
                #2  0x000055ce22768249 uv_cond_wait (heroic + 0x2213249)
                #3  0x000055ce293cf95a n/a (heroic + 0x8e7a95a)
                #4  0x000055ce293cd5a2 n/a (heroic + 0x8e785a2)
                #5  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #6  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38454:
                #0  0x00007f008aedcc5f wait4 (libc.so.6 + 0xdcc5f)
                #1  0x000055ce25806064 n/a (heroic + 0x52b1064)
                #2  0x000055ce258036c0 n/a (heroic + 0x52ae6c0)
                #3  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #4  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #5  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                
                Stack trace of thread 38448:
                #0  0x00007f008af0590f __poll (libc.so.6 + 0x10590f)
                #1  0x000055ce24ecae0f n/a (heroic + 0x4975e0f)
                #2  0x000055ce2580b544 n/a (heroic + 0x52b6544)
                #3  0x00007f008ae8cded start_thread (libc.so.6 + 0x8cded)
                #4  0x00007f008af12370 __clone3 (libc.so.6 + 0x112370)
                ELF object binary architecture: AMD x86-64
```

### Steps to reproduce

Fedora 36.

    sudo dnf copr enable atim/heroic-games-launcher
    sudo dnf install heroic

    > heroic

### Expected behavior

I expected the heroic games launcher to start.

### Heroic Version

Latest Stable

### System Information

Fedora 36
heroic-games-launcher-bin-2.5.2-1.fc36.x86_64

**Comment by marmalodak** [comment]
The issue I filed occurs when runking KDE/plasma.

When running `openbox` then heroic launches as expected.

**Comment by flavioislima** [comment]
Can you try the flatpak?

**Comment by marmalodak** [comment]
I haven't used flatpak before, hopefully this answers your question. 
Running Plasma/Wayland with flatpak, heroic runs. There are no messages about X and no core dump messages.

    ❯ flatpak run com.heroicgameslauncher.hgl
    [3:0106/165627.335481:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
    [3:0106/165627.335507:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
    (16:56:27) INFO:    [Connection]:       Connectivity: check-online
    (16:56:27) INFO:    [Connection]:       Pinging external endpoints
    (16:56:27) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
    (16:56:27) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
    (16:56:27) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
    libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
    MESA-LOADER: failed to retrieve device information
    MESA-LOADER: failed to open nvidia-drm: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/nvidia-drm_dri.so: cannot open shared object file: Permission denied (search paths /us
    failed to load driver: nvidia-drm
    MESA-LOADER: failed to open zink: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/zink_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x86_64
    failed to load driver: zink
    MESA-LOADER: failed to open kms_swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/kms_swrast_dri.so: cannot open shared object file: Permission denied (search paths /us
    failed to load driver: kms_swrast
    MESA-LOADER: failed to open swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x8
    failed to load swrast driver
    APPIMAGE env is not defined, current application is not an AppImage
    LaunchProcess: failed to execvp:
    xdg-settings
    (16:56:27) WARNING: [Backend]:          Failed to register protocol with OS.
    [3:0106/165627.474932:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    [3:0106/165627.474955:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    [3:0106/165627.474971:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    (16:56:27) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
    hostname: invalid option -- 'f'
    Try 'hostname --help' for more information.
    (16:56:27) INFO:    [Backend]:

    Heroic Version: 2.5.2 Yamato
    Legendary Version:  0.20.31 Dark Energy (hotfix #5)
    GOGdl Version: 0.5
    OS: Freedesktop SDK KERNEL: 6.0.16-200.fc36.x86_64 ARCH: x64
    CPU: Intel Gen Intel® Core™ i7-12700K @3.7 GOVERNOR: powersave
    RAM: Total: 62.56 GiB Available: 59.79 GiB
    GRAPHICS: GPU0: GA104 [GeForce RTX 3060 Ti Lite Hash Rate]  VRAM: 32MB
    PROTOCOL: wayland

    (16:56:27) INFO:    [Connection]:       Connectivity: online
    (16:56:27) INFO:    [Backend]:          User Not Found, removing it from Store
    (16:56:27) INFO:    [Backend]:          Frontend Ready
    (16:56:27) INFO:    [Backend]:          Checking for current version changelog
    (16:56:28) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
    (16:56:29) INFO:    [Winetricks]:       Downloading Winetricks
    (16:56:32) INFO:    [Backend]:          Starting the Download Queue

**Comment by marmalodak** [comment]
I haven't used flatpak before, hopefully this answers your question. 
Running Plasma/Wayland with flatpak, heroic runs. There are no messages about X and no core dump messages.

    ❯ flatpak run com.heroicgameslauncher.hgl
    [3:0106/165627.335481:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
    [3:0106/165627.335507:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
    (16:56:27) INFO:    [Connection]:       Connectivity: check-online
    (16:56:27) INFO:    [Connection]:       Pinging external endpoints
    (16:56:27) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
    (16:56:27) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
    (16:56:27) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
    libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
    MESA-LOADER: failed to retrieve device information
    MESA-LOADER: failed to open nvidia-drm: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/nvidia-drm_dri.so: cannot open shared object file: Permission denied (search paths /us
    failed to load driver: nvidia-drm
    MESA-LOADER: failed to open zink: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/zink_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x86_64
    failed to load driver: zink
    MESA-LOADER: failed to open kms_swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/kms_swrast_dri.so: cannot open shared object file: Permission denied (search paths /us
    failed to load driver: kms_swrast
    MESA-LOADER: failed to open swrast: /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/swrast_dri.so: cannot open shared object file: Permission denied (search paths /usr/lib/x8
    failed to load swrast driver
    APPIMAGE env is not defined, current application is not an AppImage
    LaunchProcess: failed to execvp:
    xdg-settings
    (16:56:27) WARNING: [Backend]:          Failed to register protocol with OS.
    [3:0106/165627.474932:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    [3:0106/165627.474955:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    [3:0106/165627.474971:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
    (16:56:27) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --version
    hostname: invalid option -- 'f'
    Try 'hostname --help' for more information.
    (16:56:27) INFO:    [Backend]:

    Heroic Version: 2.5.2 Yamato
    Legendary Version:  0.20.31 Dark Energy (hotfix #5)
    GOGdl Version: 0.5
    OS: Freedesktop SDK KERNEL: 6.0.16-200.fc36.x86_64 ARCH: x64
    CPU: Intel Gen Intel® Core™ i7-12700K @3.7 GOVERNOR: powersave
    RAM: Total: 62.56 GiB Available: 59.79 GiB
    GRAPHICS: GPU0: GA104 [GeForce RTX 3060 Ti Lite Hash Rate]  VRAM: 32MB
    PROTOCOL: wayland

    (16:56:27) INFO:    [Connection]:       Connectivity: online
    (16:56:27) INFO:    [Backend]:          User Not Found, removing it from Store
    (16:56:27) INFO:    [Backend]:          Frontend Ready
    (16:56:27) INFO:    [Backend]:          Checking for current version changelog
    (16:56:28) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
    (16:56:29) INFO:    [Winetricks]:       Downloading Winetricks
    (16:56:32) INFO:    [Backend]:          Starting the Download Queue

**Comment by flavioislima** [comment]
Yes, I think it won't work on your Fedora install due to some issues with GLIBC version (libc.so.6), so for now stick with Flatpak, it will be more stable.
